lib/main/atom/views/outline/navigationTreeComponent.tsx

```diff
@@ -0,0 +1,409 @@
+        }
+        return true
+      }
+      return false
+    }
+  }
+
+  /**
+   * HELPER select the node's HTML represenation which corresponds to the
+   *        current cursor position
+   */
+  private selectAtCursorLine(): void {
+    this.whileRendering = {
+      lastCursorLine:
+        this.editor && this.editor.getLastCursor()
+          ? this.editor.getLastCursor().getBufferRow()
+          : null,
+    }
+
+    const cursorLine = this.whileRendering.lastCursorLine
+    if (!cursorLine || !this.props.navTree || !this.refs.main) {
+      return
+    }
+
+    const selectedChild = this.findNodeAt(cursorLine, cursorLine, this.props.navTree)
+    if (selectedChild !== null) {
+      // console.log("select at cursor-line " + cursorLine, selectedChild) // DEBUG
+      this.selectedNode = selectedChild
+      etch.update(this)
+    }
+  }
+
+  /**
+   * HELPER find the node that is "furthest down" the
+   *        node hiearchy, i.e. which's start-, end-position contains the
+   *        cursorLine AND is smallest.
+   * @param  {NavigationTreeViewModel} node
+   *                  the HTML element from which to start searching
+   * @param  {Number} cursorLine the cursor line
+   * @return {HTMLElement|null} the node's HTML representation which matches cursorLine
+   *                            (i.e. which' start-, end-position contain cursorLine while
+   *                             having the smallest distance to cursorLine), or NULL if no
+   *                            matching node can be found
+   */
+  private findNodeAt(
+    startLine: number,
+    endLine: number,
+    node: NavigationTreeViewModel,
+  ): NavigationTreeViewModel | null {
+    if (!node.childItems) {
+      return null
+    }
+
+    for (const elem of node.childItems) {
+      const start: number = this.getNodeStartLine(elem)
+      const end: number = this.getNodeEndLine(elem)
+      if (isFinite(start) && isFinite(end)) {
+        if (startLine >= start && endLine <= end) {
+          const selected = this.findNodeAt(startLine, endLine, elem)
+          if (selected) {
+            return selected
+          } else {
+            return elem
+          }
+        } else if (isFinite(end) && endLine < end) {
+          break
+        }
+      }
+
+      const selectedChild = this.findNodeAt(startLine, endLine, elem)
+
+      if (selectedChild) {
+        return selectedChild
+      }
+    }
+
+    const nstart: number = this.getNodeStartLine(node)
+    const nend: number = this.getNodeEndLine(node)
+    if (isFinite(nstart) && isFinite(nend) && startLine >= nstart && endLine <= nend) {
+      return node
+    }
+    return null
+  }
+
+  /**
+   * HELPER scroll the node's HTML representation (i.e. domNode) into view
+   *        (i.e. scroll the semantic-view's tree representation)
+   * @param  {Element} domNode the HTMLElement that should be made visisble
+   */
+  private scrollTo(domNode: Element): void {
+    const elem: any = domNode
```

Comment:
:warning: Probably better to augment `Element` interface than cast to `any`

Reply:
OK (changed in next commit)